components/script/dom/windowproxy.rs

```diff
@@ -983,8 +990,10 @@ pub fn new_window_proxy_handler() -> WindowProxyHandler {
 #[allow(unsafe_code)]
 unsafe fn throw_security_error(cx: *mut JSContext) -> bool {
     if !JS_IsExceptionPending(cx) {
-        let global = GlobalScope::from_context(cx);
-        throw_dom_exception(SafeJSContext::from_ptr(cx), &*global, Error::Security);
+        let safe_context = SafeJSContext::from_ptr(cx);
+        let in_compartment_proof = AlreadyInCompartment::assert_for_cx(safe_context);
```

Comment:
Instead, let's add an `InCompartment` argument to `throw_security_error` and make the callers prove that they're in a realm.